test/framework/src/main/java/org/elasticsearch/test/TestCluster.java

```diff
@@ -136,52 +235,93 @@ public void assertAfterTest() throws Exception {
      * all indices are removed.
      */
     public void wipeIndices(String... indices) {
+        final PlainActionFuture<Void> future = new PlainActionFuture<>();
+        wipeIndicesAsync(indices, future);
+        try {
+            future.get();
```

Comment:
Could we use a `safeGet` or `safeAwait` variant? Or is 10s not enough here? In any case I'd rather a bounded wait and we should restore the interrupted flag if set.

Reply:
Good question in 2024 :) I don't know. Maybe let's add a timeout for this stuff in a follow-up? I'll make sure to set the interrupt again here though. Sec :)

Reply:
Actually scratch that ... more than 10s for a test cleanup would be stupid ... lets just use `safeGet` and fix things if this ever fails.